viewing_party/party.py

```diff
@@ -0,0 +1,170 @@
+# Wave 1
+
+def create_movie(title, genre, rating):
+    create_movie_dict = {
+        "title": title,
+        "genre": genre,
+        "rating": rating
+    }
+    if title and genre and rating:
+      return create_movie_dict 
+    else:
+      return None
+
+def add_to_watched(user_data, movie):
+    watched_list = user_data["watched"]
+    watched_list.append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    add_watchlist = user_data["watchlist"]
+    add_watchlist.append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data['watchlist']:
+        if title == movie['title']:
+            user_data['watchlist'].remove(movie)
+            user_data['watched'].append(movie)
+    return user_data
+
+
+# Wave 2
+
+def get_watched_avg_rating(user_data):
+    pass
+    if len(user_data['watched']) > 0:
+      sum_rating = 0
+      for movie in user_data['watched']:
+        rating = movie['rating']
+        sum_rating += rating 
+      
+      return sum_rating/len(user_data['watched'])
+    else:
+      return 0.0
+
+def get_most_watched_genre(user_data):
+    pass
+    genres = []
+    if len(user_data['watched']) > 0:
+        for movie in user_data['watched']:
+            for genre in movie['genre']:
+                genre = movie['genre']
+                genres.append(genre)
+    
+        most_common = max(genres, key = genres.count)
+        return most_common
+    elif len(user_data['watched']) == 0:
+        return None 
+
+# Wave 3
+
+def get_unique_watched(user_data):
+    user_list = []
+    friends_list = []
+    final_list = []
+    for movies in user_data['friends']:
+        for movie in movies['watched']:
+            friends_list.append(movie)
+    for movie in user_data['watched']:
+        user_list.append(movie)
+    for movie in user_list:
+        if movie not in friends_list:
+            final_list.append(movie)
+
+    return final_list
+def get_friends_unique_watched(user_data):
+    user_list = []
+    friends_list = []
+    final_list = []
+    
+    for movies in user_data['friends']:
+        for movie in movies['watched']:
+            friends_list.append(movie)
+    for movie in user_data['watched']:
+        user_list.append(movie)
+    for movie in friends_list:
+        if movie not in user_list and movie not in final_list:
+            final_list.append(movie)
```

Comment:
This function works well but you could do this with two lists and these lines could change to 
```suggestion
    for movie in user_data['watched']:
        if movie["title"] not in friends_list:
            final_list.append(movie)
```